language-plutus-core/src/Language/PlutusCore/Quote.hs

```diff
@@ -89,6 +97,10 @@ freshenName (Name ann str _) = Name ann str <$> freshUnique
 freshTyName :: (Monad m) => a -> BSL.ByteString -> QuoteT m (TyName a)
 freshTyName = fmap TyName .* freshName
 
+-- | Get a fresh 'TyName', given the annotation an the name.
+freshTyNameText :: (Monad m) => a -> Text.Text -> QuoteT m (TyName a)
```

Comment:
Why is this necessary?

Reply:
This is needed whenever you want to do something with the name of a variable. Like when you need to append an index to the name or, I don't know, uncapitilize it. Manipulating `ByteString`s, especially lazy, is annoying in this regard.